### PR TITLE
net-vpn/networkmanager-l2tp: bump to 1.20.8, improvements

### PR DESCRIPTION
**Pull request by kuraga**
1.20.6 is out: https://github.com/nm-l2tp/NetworkManager-l2tp/releases/tag/1.20.6

Also, net-misc/networkmanager dependency has been fixed (as in [bug #876957](https://bugs.gentoo.org/show_bug.cgi?id=876957) / #27764).

Closes: https://bugs.gentoo.org/884461

### CODE REVIEW

File: net-vpn/networkmanager-l2tp/networkmanager-l2tp-1.20.6.ebuild [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+	dev-util/intltool
+	virtual/pkgconfig"
+
+S="${WORKDIR}/${MY_P}"
+
+src_prepare() {
+	default
+
+	eautoreconf
+}
```
Comment (juippis) [inline net-vpn/networkmanager-l2tp/networkmanager-l2tp-1.20.6.ebuild:47]: There doesn't seem to be a reason to define this phase, default should be enough.

File: net-vpn/networkmanager-l2tp/networkmanager-l2tp-1.20.6.ebuild [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+src_configure() {
+	local PPPD_VER=$(best_version net-dialup/ppp)
+	PPPD_VER=${PPPD_VER#*/*-} # reduce it to ${PV}-${PR}
+	PPPD_VER=${PPPD_VER%%[_-]*} # main version without beta/pre/patch/revision
+
+	local myeconfargs=(
+		--localstatedir=/var
+		--with-pppd-plugin-dir=/usr/$(get_libdir)/pppd/${PPPD_VER}
+		$(use_with gtk gnome)
+		$(use_enable static-libs static)
```
Comment (juippis) [inline net-vpn/networkmanager-l2tp/networkmanager-l2tp-1.20.6.ebuild:58]: Is there a reason to install static libs for this package? I'd encourage to just disable these unconditionally.

File: net-vpn/networkmanager-l2tp/networkmanager-l2tp-1.20.6.ebuild [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+	)
+	gtk? (
+		x11-libs/gtk+:3
+		app-crypt/libsecret
+		gnome-extra/nm-applet
+	)"
+DEPEND="${RDEPEND}"
+BDEPEND="
+	sys-devel/gettext
+	dev-util/intltool
```
Comment (juippis) [inline net-vpn/networkmanager-l2tp/networkmanager-l2tp-1.20.6.ebuild:38]: Sort alphabetically please.

File: net-vpn/networkmanager-l2tp/networkmanager-l2tp-1.20.6.ebuild [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="gtk static-libs"
+
+RDEPEND="
+	>=net-misc/networkmanager-1.20[ppp]
+	dev-libs/dbus-glib
+	net-dialup/ppp:=[eap-tls]
+	net-dialup/xl2tpd
+	>=dev-libs/glib-2.32
```
Comment (juippis) [inline net-vpn/networkmanager-l2tp/networkmanager-l2tp-1.20.6.ebuild:25]: Sort alphabetically please.

**Review by juippis — COMMENTED** [review]
It also looks like you need to add some missing deps, 
```
 * VDB: detected possibly incorrect RDEPEND (net-vpn/networkmanager-l2tp-1.20.6)
 * dev-libs/dbus-glib < 
 * dev-libs/glib      | dev-libs/glib:2
 *                    > dev-libs/nspr
 *                    > dev-libs/nss
 *                    > dev-libs/openssl:=
 *                    > net-libs/libnma
```

I think #27764 can be closed if this is to be merged, then we can clean out older versions immediately.

**Comment by kuraga** [comment]
> It also looks like you need to add some missing deps,
> 
> ```
>  * VDB: detected possibly incorrect RDEPEND (net-vpn/networkmanager-l2tp-1.20.6)
>  * dev-libs/dbus-glib < 
>  * dev-libs/glib      | dev-libs/glib:2
>  *                    > dev-libs/nspr
>  *                    > dev-libs/nss
>  *                    > dev-libs/openssl:=
>  *                    > net-libs/libnma
> ```
> 
> I think #27764 can be closed if this is to be merged, then we can clean out older versions immediately.

@juippis , where can I see it? How do I have to deal with it?

File: net-vpn/networkmanager-l2tp/networkmanager-l2tp-1.20.6.ebuild (hunk 1 above)
Comment (kuraga) [reply net-vpn/networkmanager-l2tp/networkmanager-l2tp-1.20.6.ebuild]: Fixed. Removed `inherit autotools`.

File: net-vpn/networkmanager-l2tp/networkmanager-l2tp-1.20.6.ebuild (hunk 2 above)
Comment (kuraga) [reply net-vpn/networkmanager-l2tp/networkmanager-l2tp-1.20.6.ebuild]: Fixed.

File: net-vpn/networkmanager-l2tp/networkmanager-l2tp-1.20.6.ebuild (hunk 3 above)
Comment (kuraga) [reply net-vpn/networkmanager-l2tp/networkmanager-l2tp-1.20.6.ebuild]: Fixed.

File: net-vpn/networkmanager-l2tp/networkmanager-l2tp-1.20.6.ebuild (hunk 4 above)
Comment (kuraga) [reply net-vpn/networkmanager-l2tp/networkmanager-l2tp-1.20.6.ebuild]: Fixed.

**Comment by kuraga** [comment]
Force pushed. Also, moved to 1.20.8 version.

**Comment by thesamesam** [comment]
I'd pushed the earlier version of this after merging it last night locally, reopening.

**Comment by kuraga** [comment]
Thanks!

**Review by juippis — APPROVED** [review]
Few missing deps, I'll add. Thanks!

**Comment by kuraga** [comment]
@juippis , thanks! But say, please, which instrument it was? Couldn't google it :( Isn't it `pkgcheck`, is it?
```
 * VDB: detected possibly incorrect RDEPEND (net-vpn/networkmanager-l2tp-1.20.6)
```

**Comment by juippis** [comment]
Hey, 

it comes from
https://packages.gentoo.org/packages/app-portage/iwdevtools

Documentation
https://github.com/ionenwks/iwdevtools#qa-vdb

**Comment by kuraga** [comment]
Thanks :)

**Comment by kuraga** [comment]
@juippis , one more, please.

1. In 8160a74439077c8f81527ba52c63637d6ca813a7, you added more `RDEPEND`s & `BDEPEND`s than https://github.com/gentoo/gentoo/pull/28548#pullrequestreview-1264834036 / `qa-vdb` / `qa-vdb --depend` show. How did you find them?

2. Also IMHO we have to backport b8ce8a793839b5a2c13c785da5c6d727625db3e0, 922945c527bf1974135f036f6ada5ef77da854ea, e8d34a8da7e755cad71678c7cf993369ae5b3562 and 8160a74439077c8f81527ba52c63637d6ca813a7 on 1.8.8 version: created #29247 (but p.1's answer can change `RDEPEND`s & `BDEPEND` for `[gtk]`).

**Comment by kuraga** [comment]
> > ```
> > 1. In [8160a74](https://github.com/gentoo/gentoo/commit/8160a74439077c8f81527ba52c63637d6ca813a7), you added more `RDEPEND`s & `BDEPEND`s than [net-vpn/networkmanager-l2tp: bump to 1.20.8, improvements #28548 (review)](https://github.com/gentoo/gentoo/pull/28548#pullrequestreview-1264834036) / `qa-vdb` / `qa-vdb --depend` show. How did you find them?
> > ```
> 
> The DEPEND and BDEPEND changes were found by emering the package on a clean stage3 container, with no extra packages installed.
> 
> The RDEPEND changes were detected by **qa-vdb** from *_iwdevtools_. Toggle the `gtk` use flag on/off to get precise results.

@juippis , thanks very much for your advice!

With gratitude, I hurry up though to warn that I found `qa-vdb`'s result incomplete :) https://github.com/ionenwks/iwdevtools/issues/15

**Pull request closed without merging**